File: AndroidAssignment4/app/src/main/java/com/example/androidassignment4/MyAdapter.java
package com.example.androidassignment4;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    private ArrayList<String> data;

    //Constructor of the Adaptor Class.
    public MyAdapter(ArrayList<String> data){
        this.data = data;
    }

    //onCreateViewHolder, onBindViewHolder, getItemCount methods Implemented
    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        //inflating the single item view layout.
        View nameText = inflater.inflate(R.layout.question1_layout, parent, false);
        return new MyViewHolder(nameText);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        //binding data to the layout.
        holder.nameText.setText(data.get(position));
    }


    @Override
    public int getItemCount() {
        return data.size();
    }


    //Custom View Holder Class to hold the Elements in the single item layout file
    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView nameText;
        public MyViewHolder(@NonNull View itemView){
            super(itemView);
            nameText = itemView.findViewById(R.id.nameView);
        }
    }

}

File: Gallerio/app/src/main/java/com/examples/gallerio/MyAdapter.kt
package com.examples.gallerio

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class MyAdapter(val mDatasetSample: ArrayList<String>, val contex: Context) : RecyclerView.Adapter<MyAdapter.ViewHolder>() {

//    var imageUrl: ArrayList<CategoryModel> = mDatasetSample

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        return ViewHolder(LayoutInflater.from(contex).inflate(R.layout.timeline_item_view, parent, false))
//        return ViewHolder(LayoutInflater.from(contex).inflate(R.layout.single_item, parent, false))
    }

    override fun getItemCount(): Int {
        return mDatasetSample.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        //binding data to the layout.
        holder.titleText.text = mDatasetSample.get(position)
      //  holder.titleImage.setImageResource(mImage)


    }

    class ViewHolder (view: View) : RecyclerView.ViewHolder(view) {
        var titleText = view.findViewById(R.id.name) as TextView
       // var titleImage = view.findViewById(R.id.timelineImage) as ImageView

    }


}
